opentech/apply/users/urls.py

```diff
@@ -2,15 +2,60 @@
 
 from django.contrib.auth import views as auth_views
 
+from django.urls import reverse_lazy
+
+from wagtail.wagtailadmin.forms import PasswordResetForm
+
+from opentech.apply.users.views import account
+
 urlpatterns = [
-    # TODO replace with dashboard view with a login_required decorator
-    url(r'^$', auth_views.login, {
-        'template_name': 'users/login.html'
-    }, name='user'),
-    url(r'^login$', auth_views.login, {
-        'template_name': 'users/login.html'
-    }, name='login'),
+    url(r'^$', account, name='account'),
+    url(
+        r'^login/$',
+        auth_views.LoginView.as_view(
+            template_name='users/login.html',
+            redirect_authenticated_user=True
+        ),
+        name='login'
+    ),
 
     # Log out
-    url(r'^logout/$', auth_views.logout, {'next_page': '/'}, name='logout'),
+    url(r'^logout/$', auth_views.LogoutView.as_view(next_page='/'), name='logout'),
+
+    # Password change
+    url(
+        r'^password/$',
+        auth_views.PasswordChangeView.as_view(
+            template_name="users/change_password.html",
+            success_url=reverse_lazy('users:account')
+        ),
+        name='password_change',
+    ),
+
+    # Password reset
+    url(r'^reset/$', auth_views.PasswordResetView.as_view(
+        template_name='users/password_reset/form.html',
+        email_template_name='users/password_reset/email.txt',
+        success_url=reverse_lazy('users:password_reset_done'),
+        form_class=PasswordResetForm
```

Comment:
I'm not a fan of the wagtail password reset form, it leaks information about whether the user is registered with the site.

I would prefer that we accept all emails and redirect to a page with "email sent to: {{ email }} with further instructions."  which i think is what the standard django form does. 
https://docs.djangoproject.com/en/2.0/topics/auth/default/#django.contrib.auth.forms.PasswordResetForm 

https://github.com/django/django/blob/master/django/contrib/auth/forms.py#L285 Only checks you enter something that validates as an email address.

Reply:
👍 
worth raising a Wagtail issue too